client/landing/stepper/declarative-flow/internals/components/step-route/index.tsx

```diff
@@ -41,7 +50,13 @@ const StepRoute = ( { step, flow, showWooLogo, renderStep, navigate }: StepRoute
 	}, [ loginUrl, shouldAuthUser, useBuiltItInAuth ] );
 
 	if ( useBuiltItInAuth && shouldAuthUser && ! userIsLoggedIn ) {
-		navigate( 'user', undefined, true );
+		// If the current step requires the auth, it should become a next step after the auth.
+		const extraData = {
+			previousStep: stepData?.previousStep,
+			nextStep: step.slug,
+		};
```

Comment:
I think the extraData feature is a source of bugs and I was hoping we should remove it. Luckily, only 2-3 flows use it at this point. Adding state in general should be an elaborate and careful design. The main reason I prefer Stepper to Start is the statelessness. It would be amazing if we can come up with another way.

Reply:
What's the bug of the `extraData`? I agreed we can remove it if it's not a great design or re-design the structure of the step data. However, I think it's an urgent issue for the onboarding flow and we can come up with a better solution in the future.

Reply:
The problem is with uncontrolled and untyped state in general. You can see that affecting start a lot.

Extra data makes steps aware of each other and potentially depend on each other. Steps should be like fancy form inputs, completely stupid and oblivious about their position in the flow. They just submit some data and the flow decides what happens. This makes the steps much more robust and reusable in every flow.